File: wup1/Java/Front22.java
/* 
    front22

    Given a string, take the first 2 chars and return the string with the 2 chars 
    added at both the front and back, so "kitten" yields"kikittenki". 
    If the string length is less than 2, use whatever chars are there.
*/

public class Front22 {
    
    public static void main(String[] args) {
        Front22 f22 = new Front22();
        f22.test();
	}

    public String front22(String str) {
        String start ;
        if (str.length() >= 2) 
            start = str.substring(0, 2);
        else
            start = str;
        return start + str + start;
    }

    public void test() {
        assert front22("kitten").equals("kikittenki");
        assert front22("Ha").equals("HaHaHa");
        assert front22("abc").equals("ababcab");
        assert front22("ab").equals("ababab");	
        assert front22("a").equals("aaa");	
        assert front22("").equals("");	
        assert front22("Logic").equals("LoLogicLo");
        System.out.println("All tests passed");
    }
}

File: wup1/JS/front22.js
/* 
    Warmup-1 -- front22
    Given a string, take the first 2 chars and return the string with the 2 chars added at both the front and back, so 'kitten' yields 'kikittenki' chars are there.

    Examples

    front22('kitten'), 'kikittenki
    front22('Ha'), 'HaHaHa
    front22('abc'), 'ababcab
*/

function front22(str){
  return str.substring(0,2) + str + str.substring(0,2);
}

const assert = require('assert');
assert.equal(front22('kitten'), 'kikittenki');
assert.equal(front22('Ha'), 'HaHaHa');
assert.equal(front22('abc'), 'ababcab');
assert.equal(front22('ab'), 'ababab');	
assert.equal(front22('a'), 'aaa');	
assert.equal(front22(''), ''); 	
assert.equal(front22('Logic'), 'LoLogicLo');
console.log("All tests passed");